tests/test_cross_validation.py

```diff
@@ -248,98 +248,38 @@ def test_stratified_split(implicit_interactions_to_split,
 ):
-    train_expected_df = pd.DataFrame(
-        data={
-            'user_id': [0, 1, 1, 1, 1, 1, 2, 2, 3, 3, 4, 4],
-            'item_id': [0, 2, 3, 4, 6, 8, 3, 4, 1, 2, 1, 5],
-            'rating': [1, 3, 4, 5, 3, 1, 3, 4, 1, 3, 2, 4],
-        }
-    )
-    train_expected = Interactions(
-        mat=coo_matrix(
-            (
-                train_expected_df['rating'],
-                (train_expected_df['user_id'], train_expected_df['item_id']),
-            ),
-            shape=(interactions_to_split_with_a_user_with_only_one_interaction.num_users,
-                   interactions_to_split_with_a_user_with_only_one_interaction.num_items),
-        ),
-        allow_missing_ids=True,
-        check_num_negative_samples_is_valid=False,
-    )
-
-    validate_expected_df = pd.DataFrame(
-        data={
-            'user_id': [1, 2, 3, 4],
-            'item_id': [7, 2, 3, 4],
-            'rating': [2, 2, 4, 5],
-        }
-    )
-    validate_expected = Interactions(
-        mat=coo_matrix(
-            (
-                validate_expected_df['rating'],
-                (validate_expected_df['user_id'], validate_expected_df['item_id']),
-            ),
-            shape=(interactions_to_split_with_a_user_with_only_one_interaction.num_users,
-                   interactions_to_split_with_a_user_with_only_one_interaction.num_items),
-        ),
-        allow_missing_ids=True,
-        check_num_negative_samples_is_valid=False,
-    )
-
-    test_expected_df = pd.DataFrame(
-        data={
-            'user_id': [1, 1, 2, 3, 4],
-            'item_id': [1, 5, 1, 4, 2],
-            'rating': [2, 4, 1, 5, 3],
-        }
-    )
-    test_expected = Interactions(
-        mat=coo_matrix(
-            (
-                test_expected_df['rating'],
-                (test_expected_df['user_id'], test_expected_df['item_id']),
-            ),
-            shape=(interactions_to_split_with_a_user_with_only_one_interaction.num_users,
-                   interactions_to_split_with_a_user_with_only_one_interaction.num_items),
-        ),
-        allow_missing_ids=True,
-        check_num_negative_samples_is_valid=False,
-    )
+    users_with_only_one_interaction = [0, 5, 6]
 
-    (train_actual, validate_actual, test_actual) = stratified_split(
-        interactions=interactions_to_split_with_a_user_with_only_one_interaction,
+    (train_actual, _, _) = stratified_split(
+        interactions=interactions_to_split_with_users_with_only_one_interaction,
         val_p=0.1,
         test_p=0.2,
         seed=42,
-        ignore_warning=True
+        force_split=True
     )
 
-    np.testing.assert_array_equal(train_actual.toarray(), train_expected.toarray())
-    np.testing.assert_array_equal(
-        validate_actual.toarray(), validate_expected.toarray()
-    )
-    np.testing.assert_array_equal(test_actual.toarray(), test_expected.toarray())
+    assert all(user in train_actual[:][0][0].tolist() for user in users_with_only_one_interaction)
```

Comment:
Nice!!!!!!!

Reply:
I wasn't sure if you'd like this one, glad you do! :D

Reply:
I wasn't sure you'd like this one, glad you do! :D